File: go/1402.go
package main

import "sort"

func maxSatisfaction(satisfaction []int) int {
	sort.Slice(satisfaction, func(i, j int) bool {
		return satisfaction[i] > satisfaction[j]
	})
	ans := 0
	total := 0
	for _, score := range satisfaction {
		total += score
		if total > 0 {
			ans += total
		} else {
			break
		}
	}
	return ans

}

func main() {

}

File: python/1402.py
from typing import List

class Solution:
    def maxSatisfaction(self, satisfaction: List[int]) -> int:
        satisfaction.sort(reverse=True)
        total = 0
        ans = 0
        for score in satisfaction:
            if score + total > 0:
                ans += total + score
                total += score
            else: break
        return ans

satisfaction = [-1,-8,0,5,-9]
s= Solution()
print(s.maxSatisfaction(satisfaction))